contracts/KIP/access/KAccessControl.sol

```diff
@@ -1,16 +1,248 @@
 
 pragma solidity ^0.8.0;
 
-import "../../access/AccessControl.sol";
+import "./IKAccessControl.sol";
+import "../../utils/Context.sol";
+import "../../utils/Strings.sol";
 import "../utils/introspection/KIP13.sol";
 
-abstract contract KAccessControl is KIP13, AccessControl {
-    function supportsInterface(bytes4 interfaceId) public view virtual override(KIP13, AccessControl) returns (bool) {
-        return super.supportsInterface(interfaceId);
+/**
+ * @dev Contract module that allows children to implement role-based access
+ * control mechanisms. This is a lightweight version that doesn't allow enumerating role
+ * members except through off-chain means by accessing the contract event logs. Some
+ * applications may benefit from on-chain enumerability, for those cases see
+ * {AccessControlEnumerable}.
+ *
+ * Roles are referred to by their `bytes32` identifier. These should be exposed
+ * in the external API and be unique. The best way to achieve this is by
+ * using `public constant` hash digests:
+ *
+ * ```
+ * bytes32 public constant MY_ROLE = keccak256("MY_ROLE");
+ * ```
+ *
+ * Roles can be used to represent a set of permissions. To restrict access to a
+ * function call, use {hasRole}:
+ *
+ * ```
+ * function foo() public {
+ *     require(hasRole(MY_ROLE, msg.sender));
+ *     ...
+ * }
+ * ```
+ *
+ * Roles can be granted and revoked dynamically via the {grantRole} and
+ * {revokeRole} functions. Each role has an associated admin role, and only
+ * accounts that have a role's admin role can call {grantRole} and {revokeRole}.
+ *
+ * By default, the admin role for all roles is `DEFAULT_ADMIN_ROLE`, which means
+ * that only accounts with this role will be able to grant or revoke other
+ * roles. More complex role relationships can be created by using
+ * {_setRoleAdmin}.
+ *
+ * WARNING: The `DEFAULT_ADMIN_ROLE` is also its own admin: it has permission to
+ * grant and revoke this role. Extra precautions should be taken to secure
+ * accounts that have been granted it.
+ */
+abstract contract KAccessControl is Context, IKAccessControl, KIP13 {
+    struct RoleData {
+        mapping(address => bool) members;
+        bytes32 adminRole;
+    }
+
+    mapping(bytes32 => RoleData) private _roles;
+
+    bytes32 public constant DEFAULT_ADMIN_ROLE = 0x00;
+
+    /**
+     * @dev Modifier that checks that an account has a specific role. Reverts
+     * with a standardized message including the required role.
+     *
+     * The format of the revert reason is given by the following regular expression:
+     *
+     *  /^AccessControl: account (0x[0-9a-f]{40}) is missing role (0x[0-9a-f]{64})$/
+     *
+     * _Available since v4.1._
+     */
+    modifier onlyRole(bytes32 role) {
+        _checkRole(role);
+        _;
+    }
+
+    /**
+     * @dev See {IKIP13-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
```

Comment:
@naveed-klaytn  isn't required to override `KIP13, AccessControl` here

Reply:
Everything else Looks good to me

Reply:
@toniya-klaytn here overriding `KIP13` by default & `AccessControl` override not required because KAccessControl is **not inheriting it**. i.e:
`abstract contract KAccessControl is Context, IKAccessControl, KIP13 {`

Reply:
`KAccessControl` is based on `AccessControl` not inheriting it directly.